src/Microsoft.TestPlatform.CrossPlatEngine/Utilities/TestSourceDeterminer.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Utilities
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Test sources determiner utility class
+    /// </summary>
+    internal class TestSourceDeterminer
```

Comment:
add UTs for this.
1. Multiple sources in adapterSourceMap should aggregate
2. If duplicate tests sources given, only distinct test sources should be in the list.
3. Null check tests for both GetSources and GetDefaultCodeBasePath